ibis/expr/tests/test_newrels.py

```diff
@@ -32,6 +35,19 @@
 )
 
 
+@contextlib.contextmanager
+def self_references(*tables):
+    old_counter = ops.SelfReference._uid_counter
+    # set a new counter with 1000 to avoid colliding with manually created
+    # self-references using t.view()
+    new_counter = itertools.count(1000)
```

Comment:
Is 1000 significant in any way? Why not leave it out entirely?

Reply:
`t.view()` constructs a `SelfReference()` and this prevent colliding with it.

Reply:
If we start the counting again, then it is easy to accidentally generate identical references which makes the testing more error prone.

Reply:
Ah, I misunderstood what the input argument to `count` does: it's the starting point of the counting. I thought it was the upper bound.

Carry on!